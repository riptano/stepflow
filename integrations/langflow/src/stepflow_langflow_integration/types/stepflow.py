"""Stepflow workflow and step type definitions."""

from dataclasses import dataclass, field
from typing import Dict, Any, List, Optional


@dataclass
class StepflowStep:
    """Represents a Stepflow workflow step."""
    
    id: str
    component: str
    input: Dict[str, Any]
    output_schema: Optional[Dict[str, Any]] = None
    
    def to_dict(self) -> Dict[str, Any]:
        """Convert step to dictionary representation."""
        result = {
            "id": self.id,
            "component": self.component,
            "input": self.input,
        }
        
        if self.output_schema:
            result["output_schema"] = self.output_schema
        
        return result


@dataclass 
class StepflowWorkflow:
    """Represents a Stepflow workflow."""
    
    name: str
    steps: List[StepflowStep]
    description: Optional[str] = None
    version: str = "1.0"
    schema: str = "https://stepflow.org/schemas/v1/flow.json"
    output: Optional[Dict[str, Any]] = None
    
    def to_dict(self) -> Dict[str, Any]:
        """Convert workflow to dictionary representation."""
        result = {
            "schema": self.schema,
            "name": self.name,
            "steps": [step.to_dict() for step in self.steps],
        }
        
        if self.description:
            result["description"] = self.description
        
        if self.version != "1.0":
            result["version"] = self.version
        
        if self.output:
            result["output"] = self.output
        
        return result