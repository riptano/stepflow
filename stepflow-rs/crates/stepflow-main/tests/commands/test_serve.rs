use super::stepflow;
use insta_cmd::assert_cmd_snapshot;

#[test]
fn test_serve_help() {
    assert_cmd_snapshot!(stepflow().arg("serve").arg("--help"));
}
