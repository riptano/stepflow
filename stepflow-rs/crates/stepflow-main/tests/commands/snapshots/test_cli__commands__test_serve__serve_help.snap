---
source: crates/stepflow-main/tests/commands/test_serve.rs
info:
  program: stepflow
  args:
    - "--log-file=/dev/null"
    - "--omit-stack-trace"
    - serve
    - "--help"
---
success: true
exit_code: 0
----- stdout -----
Start a StepFlow service

Usage: stepflow serve [OPTIONS]

Options:
      --log-level <LEVEL>
          Set the log level for StepFlow
          
          [default: info]
          [possible values: trace, debug, info, warn, error]

      --port <PORT>
          Port to run the service on
          
          [default: 7837]

      --config <FILE>
          The path to the stepflow config file.
          
          If not specified, will look for `stepflow-config.yml` in the directory containing the workflow file. If that isn't found, will also look in the current directory.

      --other-log-level <LEVEL>
          Set the log level for other parts of StepFlow
          
          [default: warn]
          [possible values: trace, debug, info, warn, error]

      --log-file <FILE>
          Write logs to a file instead of stderr

      --omit-stack-trace
          Omit stack traces (line numbers of errors)

  -h, --help
          Print help (see a summary with '-h')

----- stderr -----
